RecoTracker/TkDetLayers/src/Phase2EndcapSubDisk.h

```diff
@@ -0,0 +1,63 @@
+#ifndef TkDetLayers_Phase2EndcapSubDisk_h
+#define TkDetLayers_Phase2EndcapSubDisk_h
+
+#include "TrackingTools/DetLayers/interface/RingedForwardLayer.h"
+#include "Phase2EndcapSingleRing.h"
+#include "TkDetUtil.h"
+#include <array>
+#include <atomic>
+
+/** A concrete implementation for Phase 2 Endcap/Forward layer 
+ *  built out of Phase2EndcapSingleRings
+ *  this classs is used for both OT and Pixel detector
+ */
+
+#pragma GCC visibility push(hidden)
+class Phase2EndcapSubDisk final : public RingedForwardLayer {
+public:
+  Phase2EndcapSubDisk(std::vector<const Phase2EndcapSingleRing*>& rings) __attribute__((cold));
```

Comment:
for my education, why is `((cold))` used here and elsewhere in these classes?

Reply:
I don't really know what this does - this again follows the already existing classes that have these attributes.

Reply:
OK, here's what's in the manual:
```
cold
    The cold attribute is used to inform the compiler that a function is unlikely executed. The function is optimized for size rather than speed and on many targets it is placed into special subsection of the text section so all cold functions appears close together improving code locality of non-cold parts of program. The paths leading to call of cold functions within code are marked as unlikely by the branch prediction mechanism. It is thus useful to mark functions used to handle unlikely conditions, such as perror, as cold to improve optimization of hot functions that do call marked functions in rare occasions. 
```
does it apply here?

Reply:
mmmmh, unclear if that's actually the case here (even after reading it 3 times I'm not fully sure I get it :-) ). Have removed the attributes for now.